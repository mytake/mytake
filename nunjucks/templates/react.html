{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
