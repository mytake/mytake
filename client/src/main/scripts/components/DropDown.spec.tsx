import * as React from "react";
import * as renderer from "react-test-renderer";
import DropDown from "./DropDown";

const Toggle = <span>toggle text</span>;
const Children = (
  <div>
    <p>Some child</p>
  </div>
);

test("DropDown BL", () => {
  const tree = renderer
    .create(
      <DropDown
        children={Children}
        classModifier={"searchbar"}
        dropdownPosition="BL"
        toggleText={Toggle}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("DropDown disabled", () => {
  const tree = renderer
    .create(
      <DropDown
        children={Children}
        classModifier={"searchbar"}
        dropdownPosition="BL"
        disabled={true}
        toggleText={Toggle}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("DropDown enabled", () => {
  const tree = renderer
    .create(
      <DropDown
        children={Children}
        classModifier={"searchbar"}
        dropdownPosition="BL"
        disabled={false}
        toggleText={Toggle}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("DropDown BR", () => {
  const tree = renderer
    .create(
      <DropDown
        children={Children}
        classModifier={"searchbar"}
        dropdownPosition="BR"
        toggleText={Toggle}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("DropDown TL", () => {
  const tree = renderer
    .create(
      <DropDown
        children={Children}
        classModifier={"searchbar"}
        dropdownPosition="TL"
        toggleText={Toggle}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("DropDown TR", () => {
  const tree = renderer
    .create(
      <DropDown
        children={Children}
        classModifier={"searchbar"}
        dropdownPosition="TR"
        toggleText={Toggle}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
