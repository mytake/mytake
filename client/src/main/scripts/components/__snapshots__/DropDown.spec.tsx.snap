// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown BL 1`] = `
<div
  className="dropdown dropdown--searchbar"
>
  <button
    className="dropdown__toggle dropdown__toggle--searchbar"
    onClick={[Function]}
  >
    <span>
      toggle text
    </span>
  </button>
  <div
    className="dropdown__dropdown dropdown__dropdown--hidden"
  >
    <div>
      <p>
        Some child
      </p>
    </div>
  </div>
</div>
`;

exports[`DropDown BR 1`] = `
<div
  className="dropdown dropdown--searchbar"
>
  <button
    className="dropdown__toggle dropdown__toggle--searchbar"
    onClick={[Function]}
  >
    <span>
      toggle text
    </span>
  </button>
  <div
    className="dropdown__dropdown dropdown__dropdown--hidden"
  >
    <div>
      <p>
        Some child
      </p>
    </div>
  </div>
</div>
`;

exports[`DropDown TL 1`] = `
<div
  className="dropdown dropdown--searchbar"
>
  <button
    className="dropdown__toggle dropdown__toggle--searchbar"
    onClick={[Function]}
  >
    <span>
      toggle text
    </span>
  </button>
  <div
    className="dropdown__dropdown dropdown__dropdown--hidden"
  >
    <div>
      <p>
        Some child
      </p>
    </div>
  </div>
</div>
`;

exports[`DropDown TR 1`] = `
<div
  className="dropdown dropdown--searchbar"
>
  <button
    className="dropdown__toggle dropdown__toggle--searchbar"
    onClick={[Function]}
  >
    <span>
      toggle text
    </span>
  </button>
  <div
    className="dropdown__dropdown dropdown__dropdown--hidden"
  >
    <div>
      <p>
        Some child
      </p>
    </div>
  </div>
</div>
`;

exports[`DropDown disabled 1`] = `
<div
  className="dropdown dropdown--searchbar"
>
  <button
    className="dropdown__toggle dropdown__toggle--searchbar"
    onClick={[Function]}
  >
    <span>
      toggle text
    </span>
  </button>
  <div
    className="dropdown__dropdown dropdown__dropdown--hidden"
  >
    <div>
      <p>
        Some child
      </p>
    </div>
  </div>
</div>
`;

exports[`DropDown enabled 1`] = `
<div
  className="dropdown dropdown--searchbar"
>
  <button
    className="dropdown__toggle dropdown__toggle--searchbar"
    onClick={[Function]}
  >
    <span>
      toggle text
    </span>
  </button>
  <div
    className="dropdown__dropdown dropdown__dropdown--hidden"
  >
    <div>
      <p>
        Some child
      </p>
    </div>
  </div>
</div>
`;
